### PR TITLE
Generify hook-related cache invalidation

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

As the forthcoming notifications will be dispatched according to hook events, there should ideally be a distinction between cache- and notification-related hook logic, e.g. the notification logic not be implemented within the "cache-hooks".

This refactors the route "cache-hooks" to be a more generic, also adding distinction between Transaction Service and Config. Service events.

## Changes

- Rename `cache-hooks` to `hook` and the associated controller, service(s) and module accordingly
- Separate `EventType` into `TransactionEventType` and `ConfigEventType` and propagate changes

### CODE REVIEW

File: src/routes/hooks/entities/schemas/web-hook.schema.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { ModuleTransactionEventSchema } from '@/routes/hooks/entities/schemas/module-transaction.schema';
+import { NewConfirmationEventSchema } from '@/routes/hooks/entities/schemas/new-confirmation.schema';
+import { NewMessageConfirmationEventSchema } from '@/routes/hooks/entities/schemas/new-message-confirmation.schema';
+import { OutgoingEtherEventSchema } from '@/routes/hooks/entities/schemas/outgoing-ether.schema';
+import { OutgoingTokenEventSchema } from '@/routes/hooks/entities/schemas/outgoing-token.schema';
+import { PendingTransactionEventSchema } from '@/routes/hooks/entities/schemas/pending-transaction.schema';
+import { SafeAppsUpdateEventSchema } from '@/routes/hooks/entities/schemas/safe-apps-update.schema';
+import { SafeCreatedEventSchema } from '@/routes/hooks/entities/schemas/safe-created.schema';
+
+export const WebHookSchema = z.discriminatedUnion('type', [
```
Comment (hectorgomezv) [inline src/routes/hooks/entities/schemas/web-hook.schema.ts:17]: Should we also rename this to `EventSchema` to stay consistent?

File: src/routes/hooks/hooks.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { IQueuesRepository } from '@/domain/queues/queues-repository.interface';
+import { WebHookSchema } from '@/routes/hooks/entities/schemas/web-hook.schema';
+import { OnModuleInit } from '@nestjs/common';
+import { ConsumeMessage } from 'amqplib';
+import { Event } from '@/routes/hooks/entities/event.entity';
+import { ILoggingService } from '@/logging/logging.interface';
+
+export abstract class HooksService implements OnModuleInit {
```
Comment (hectorgomezv) [inline src/routes/hooks/hooks.service.ts:9]: While I'm totally aligned with the purpose of this refactor, I think I don't fully understand this abstract class part: are we planning to extend `HooksService` with another derived class for notifications (e.g.: `HooksNotificationsService`)? 

If that is the case, wouldn't that make `onModuleInit` execute twice, and therefore create two queue subscriptions? Would the `HooksController` dispatch the events to the proper implementation of `HooksService`, or all events would be consumed by all the implementation classes?

**Review by hectorgomezv — COMMENTED** [review]
Superb job! 👏🏻 

I've just left minor comments/thoughts.

File: src/routes/hooks/hooks.service.ts (hunk 2 above)
Comment (iamacook) [reply src/routes/hooks/hooks.service.ts:9]: As per our discussion, I've [opened a different PR](https://github.com/safe-global/safe-client-gateway/pull/1774) that moves the queue subscription into the domain, but accepts multiple listeners for said subscription. This should prevent multiple subscriptions. I will rebase this PR after merging that.

File: src/routes/hooks/entities/schemas/web-hook.schema.ts (hunk 1 above)
Comment (iamacook) [reply src/routes/hooks/entities/schemas/web-hook.schema.ts]: Renamed when rebasing according to https://github.com/safe-global/safe-client-gateway/pull/1770#discussion_r1682730520.

**Review by hectorgomezv — APPROVED** [review]
👏🏻

**Review by hectorgomezv — APPROVED** [review]

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**